src/librustc_save_analysis/lib.rs

```diff
@@ -497,7 +497,7 @@ impl<'l, 'tcx: 'l> SaveContext<'l, 'tcx> {
     }
 
     pub fn get_path_data(&self, id: NodeId, path: &ast::Path) -> Option<Data> {
-        let def = self.tcx.expect_def(id);
+        let def = option_try!(self.tcx.expect_resolution(id).maybe_full_def());
```

Comment:
The reason I'm saying this is because of #37676 which will catch that case differently here and the post-HIR-lowering compiler in general stops using `PathResolution` at all (which was a "temporary" hack I introduced ages ago, and it causes serious issues, even if it may seem more convenient in some cases).